mbus/mbus-protocol.c

```diff
@@ -2438,6 +2438,622 @@ mbus_data_error_lookup(int error)
+    case 0x33:
+    case 0x34:
+    case 0x35:
+    case 0x36:
+    case 0x37:
+    case 0x38:
+    case 0x39:
+    case 0x3A:
+    case 0x3B:
+    case 0x3C:
+    case 0x3D:
+    case 0x3E:
+    case 0x3F:
+
+    case 0x40:
+    case 0x41:
+    case 0x42:
+    case 0x43:
+    case 0x44:
+    case 0x45:
+    case 0x46:
+    case 0x47:
+    case 0x48:
+    case 0x49:
+    case 0x4A:
+    case 0x4B:
+    case 0x4C:
+    case 0x4D:
+    case 0x4E:
+    case 0x4F:
+
+    case 0x52:
+    case 0x53:
+    case 0x54:
+    case 0x55:
+    case 0x56:
+    case 0x57:
+        // E011 0010 to E101 0111
+        snprintf(buff, sizeof(buff), "Reserved (0x%.2x)", vib->vife[0]);
+        break;
+    case 0x58:
+    case 0x58 + 1:
+    case 0x58 + 2:
+    case 0x58 + 3:
+        // E101 10nn
+        n = 0x03 & vib->vife[0];
+        snprintf(buff, sizeof(buff), "Flow Temperature (%s degree F)", mbus_unit_prefix(n -3));
+        break;
+    case 0x5C:
+    case 0x5C + 1:
+    case 0x5C + 2:
+    case 0x5C + 3:
+        // E101 11nn
+        n = 0x03 & vib->vife[0];
+        snprintf(buff, sizeof(buff), "Return Temperature (%s degree F)", mbus_unit_prefix(n -3));
+        break;
+    case 0x60:
+    case 0x60 + 1:
+    case 0x60 + 2:
+    case 0x60 + 3:
+        // E110 00nn
+        n = 0x03 & vib->vife[0];
+        snprintf(buff, sizeof(buff), "Temperature Difference (%s degree F)", mbus_unit_prefix(n -3));
+        break;
+    case 0x64:
+    case 0x64 + 1:
+    case 0x64 + 2:
+    case 0x64 + 3:
+        // E110 01nn
+        n = 0x03 & vib->vife[0];
+        snprintf(buff, sizeof(buff), "External Temperature (%s degree F)", mbus_unit_prefix(n -3));
+        break;
+    case 0x68:
+    case 0x69:
+    case 0x6A:
+    case 0x6B:
+    case 0x6C:
+    case 0x6D:
+    case 0x6E:
+    case 0x6F:
+        // E110 1nnn
+        snprintf(buff, sizeof(buff), "Reserved (0x%.2x)", vib->vife[0]);
+        break;
+    case 0x70:
+    case 0x70 + 1:
+    case 0x70 + 2:
+    case 0x70 + 3:
+        // E111 00nn
+        n = 0x03 & vib->vife[0];
+        snprintf(buff, sizeof(buff), "Cold / Warm Temperature Limit (%s degree F)", mbus_unit_prefix(n -3));
+        break;
+    case 0x74:
+    case 0x74 + 1:
+    case 0x74 + 2:
+    case 0x74 + 3:
+        // E111 00nn
+        n = 0x03 & vib->vife[0];
+        snprintf(buff, sizeof(buff), "Cold / Warm Temperature Limit (%s degree C)", mbus_unit_prefix(n -3));
+        break;
+    case 0x78:
+    case 0x78 + 1:
+    case 0x78 + 2:
+    case 0x78 + 3:
+    case 0x78 + 4:
+    case 0x78 + 5:
+    case 0x78 + 6:
+    case 0x78 + 7:
+        // E111 1nnn
+        n = 0x07 & vib->vife[0];
+        snprintf(buff, sizeof(buff), "cumul. count max power (%s W)", mbus_unit_prefix(n - 3));
+        break;
+    default:
+        snprintf(buff, sizeof(buff), "Unrecognized VIF 0xFB extension: 0x%.2x", vib->vife[0]);
+        break;
+    }
+    return buff;
+}
+
+const char *
+mbus_vib_unit_lookup_FD(mbus_value_information_block *vib)
```

Comment:
Please no capital letter in function names

Reply:
ok, will update

Reply:
Done